### PR TITLE
Token updates

### PR DESCRIPTION
**Pull request by 0xforestwalker**
- upgradeable contract
- transfer restriction
- role based access control

### CODE REVIEW

File: broadcast/TokenDeploy.s.sol/84531/run-1693338905.json [hunk 1]
```diff
@@ -0,0 +1,234 @@
+{
```
Comment (oxkatsudon) [inline broadcast/TokenDeploy.s.sol/84531/run-1693338905.json:1]: I usually .gitignore broadcast folder

File: script/TokenDeploy.s.sol [hunk 2]
```diff
@@ -2,18 +2,72 @@
 pragma solidity ^0.8.17;
 
 import "forge-std/Script.sol";
+import {ERC1967Proxy} from "openzeppelin-contracts/proxy/ERC1967/ERC1967Proxy.sol";
```
Comment (oxkatsudon) [inline script/TokenDeploy.s.sol:5]: foundry 🤩

**Review by midnightnomad — APPROVED** [review]

**Comment by OxCarter** [comment]
Kindly verify the contract addresses on Basescan and add the Proxy and Implementation addresses to the Readme after deployment

File: README.md [hunk 3]
```diff
@@ -8,9 +21,14 @@
...
+Base Mainnet: `source .env && forge script script/TokenDeploy.s.sol:TokenDeployScript --force --rpc-url $BASE_RPC_URL --broadcast --verify --delay 5 --verifier-url $VERIFIER_URL -vvvv`
+
+## Deployment Addresses
 
-3. Verify (if manual verification required)
-`forge verify-contract --chain-id {chain_id_here}`
\ No newline at end of file
+### Base Testnet
+proxy: `0x603b3d3851020559d0e684e7e77d4d978a317d9b`
+implementation: `0x0768b63c1e80082D7B8310470f6a1e6FcB08408F`
```
Comment (OxCarter) [inline README.md:34]: great

**Review by OxCarter — APPROVED** [review]

**Review by oxkatsudon — APPROVED** [review]

**Pull request merged**